test/src/test_pylint.py

```diff
@@ -12,36 +12,62 @@
 
 @pytest.fixture(name="source_files")
 def discover_source_files():
-    if not test.TestBase.have_test_checkout():
-        pytest.skip("no test-checkout access")
 
-    checkout = test.TestBase.locate_test_checkout()
-
-    # Fetch list of checked-in files from git.
-    cwd = os.getcwd()
-    os.chdir(checkout)
-    files = subprocess.check_output(
-        [
-            "git",
-            "ls-tree",
-            "-rz",
-            "--full-tree",
-            "--name-only",
-            "HEAD",
-        ]
-    ).decode()
-    os.chdir(cwd)
+    # Evaluate whether `path` is a python file. This assumes that the file is
+    # opened by the caller, anyway, hence it might invoke `file` to detect the
+    # file type if in doubt.
+    def is_python_script(path):
+        if path.endswith(".py"):
+            return True
 
-    # File list is separated by NULs, so split into array.
-    files = files.split('\x00')
+        mime_valid = [
+            "text/x-python",
+            "text/x-script.python",
+        ]
 
-    # Filter out all our python files (i.e., all modules and files ending in *.py)
-    modules = ("assemblers/", "runners/", "sources/", "stages/")
-    files = filter(lambda p: p.endswith(".py") or p.startswith(modules), files)
+        mime_file = subprocess.check_output(
```

Comment:
I would really prefer using `libmagic` (`python-magic`) here, it is packaged and would save us subprocesses. Since it's just a performance thing here and it only relates to running tests I'm super adamant about doing so though.

Reply:
Is this missing a "not"?

Reply:
I have not done this change, yet. It needs a container rebuild as well. I might give it a shot, though.

Reply:
> Is this missing a "not"?

It is indeed missing a `not` and if it's not packaged for RHEL it's even less interesting :)